python/scripts/process_osm.py

```diff
@@ -0,0 +1,116 @@
+    print("   -u        : Upload OSM file to PostgreSQL database using osm2pgsql with the specified style file")
+    print("               (Optional: specify style file - default.lua is used otherwise)")
+    print(f"Usage: {os.path.basename(__file__)} -b [input_file] [relation_id] [style_file]")
+    print("   -b        : Extract greatest bounding box from given relation ID of input_file and upload to PostgreSQL database using osm2pgsql.")
+    print("               (Optional: specify style file - default.lua is used otherwise)")
+
+def extract_bbox(relation_id):
+    """Function to determine bounding box"""
+    content = load_multipolygon_by_id(relation_id)
+    min_lon, min_lat, max_lon, max_lat = find_min_max(content)
+    return min_lon, min_lat, max_lon, max_lat
+
+def process_osm_command(tag, input_file, output_file):
+    if not is_valid_extension(output_file):
+        raise InvalidInputError(f"File must have one of the following extensions: osm, osm.pbf, osm.bz2")
+    if tag == '-r':
+        subprocess.run(["osmium", "renumber", input_file, "-o", output_file])
+    elif tag == '-s':
+        subprocess.run(["osmium", "sort", input_file, "-o", output_file])
+    elif tag == '-sr':
+        tmp_file = 'tmp.osm'
+        subprocess.run(["osmium", "sort", input_file, "-o", tmp_file])
+        subprocess.run(["osmium", "renumber", tmp_file, "-o", output_file])
+        os.remove(tmp_file)
+    else:
+        raise ValueError(f"Unsupported tag: {tag}")
+
+def build_osm2pgsql(config, style_file_path, input_file, coords=None):
+    command = ["osm2pgsql", "-d", config.db_name, "-U", config.username, "-W", "-H", config.db_host, 
+               "-P", str(config.db_server_port), "--output=flex", "-S", style_file_path, input_file, "-x"]
+    if coords:
+        command.extend(["-b", coords])
+    subprocess.run(command)
+
+def import_osm_to_db():
+    input_files = ["resources/to_import.osm", "resources/to_import.osm.pbf", "resources/to_import.osm.bz2"]
+    input_file = None
+    for file in input_files:
+        if os.path.exists(file) and is_valid_extension(file):
+            input_file = file
+    if not input_file:
+        raise FileNotFoundError("There is no file to import.")
+    style_file_path = "resources/lua_styles/default.lua"
+    build_osm2pgsql(config, style_file_path, input_file)
+
+if __name__ == '__main__':
+   # If no tag is used OR script is called with -h/--help
+    if len(sys.argv) < 2 or (tag:=sys.argv[1]) in ["-h", "--help"]:
+        display_help()
+
+    elif len(sys.argv) < 3:
+        raise MissingInputError(f"Insufficient arguments. Use \"{os.path.basename(__file__)} -h/--help\" for hint.")
+    elif not os.path.exists((input_file:=sys.argv[2])):
+        raise FileNotFoundError(f"File '{input_file}' does not exist.")
+    elif not is_valid_extension(input_file):
+        raise InvalidInputError(f"File must have one of the following extensions: osm, osm.pbf, osm.bz2")
+    elif tag == "-d":
```

Comment:
Please add a brief description of what each case is handling. While the error handles are quite clear from the messages, the rest is not so much.